### PR TITLE
Draw range arrays

### PR DESCRIPTION
**Pull request by shrekshao**
Getting rid of UBO cause I can't get it work correctly within a short time. 

But the sample is enough to demonstrate the feature: draw range arrays.

### CODE REVIEW

**Comment by pjcozzi** [comment]
@shrekshao merge in master so this can be auto-merged.

@trungtle let me know when you are happy with this, and I will review.

**Comment by pjcozzi** [comment]
What is the point of this sample?  How is it different than the instancing sample?

**Comment by shrekshao** [comment]
`void drawArraysInstanced(GLenum mode, GLint first, GLsizei count, GLsizei instanceCount)`

with different `GLint first`, draw arrays start with an offset in the vertex array. So the two rects are using   different vertex pos data, comparing to instancing, using the same set of vertex array.

**Comment by pjcozzi** [comment]
Ah, OK.  Are you sure about the sample name?  I think people are going to confuse it with `drawRangeElements`, which will likely not be part of WebGL 2.

**Comment by shrekshao** [comment]
- The name is directly from Christ ogl sample pack. There's another sample called `draw_range_elements`
- [drawRangeElements](https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.9) will be in WebGL 2 due to this documentation

File: samples/draw_range_arrays.html [hunk 1]
```diff
@@ -0,0 +1,114 @@
+<!DOCTYPE html>
+<!-- Ported from the OpenGL Samples Pack: github.com/g-truc/ogl-samples/blob/master/tests/gl-320-draw-instanced.cpp -->
```
Comment (pjcozzi) [inline samples/draw_range_arrays.html]: Is this link correct?

**Comment by pjcozzi** [comment]
We'll keep the name, but if folks get confused, we can consider a rename to something like `draw_instanced_first`.

**Pull request merged**